tests/middleware/test_gzip.py

```diff
@@ -76,3 +76,27 @@ async def generator(bytes, count):
     assert response.text == "x" * 4000
     assert response.headers["Content-Encoding"] == "gzip"
     assert "Content-Length" not in response.headers
+
+
+def test_gzip_ignored_for_responses_with_encoding_set(test_client_factory):
+    def homepage(request):
+        async def generator(bytes, count):
+            for index in range(count):
+                yield bytes
+
+        streaming = generator(bytes=b"x" * 400, count=10)
+        return StreamingResponse(
+            streaming, status_code=200, headers={"Content-Encoding": "br"}
+        )
+
+    app = Starlette(
+        routes=[Route("/", endpoint=homepage)],
+        middleware=[Middleware(GZipMiddleware)],
+    )
+
+    client = test_client_factory(app)
+    response = client.get("/", headers={"accept-encoding": "gzip, br"})
+    assert response.status_code == 200
+    assert response.text == "x" * 4000
+    assert response.headers["Content-Encoding"] == "br"
+    assert "Content-Length" not in response.headers
```

Comment:
Wait, this test creates a `StreamingResponse` that claims to be encoded as Brotli but actually isn't, since its body is just `b"x" * 4000`, right? How is it possible that this test passes? Doesn't the client try to process brotli automatically on the response and [fails](https://github.com/encode/httpx/blob/c0e9dba32091ad0d58ed9d46d5bafaf892964b15/httpx/_decoders.py#L118-L121)?

```
>>> import brotli
>>> decompressor = brotli.Decompressor()
>>> decompressor.process(b"x" * 4000)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
brotli.error: BrotliDecoderDecompressStream failed while processing the stream
```

Reply:
I've created what I think is a better test [here](https://github.com/fullonic/brotli-asgi/pull/24/files#diff-f455f302936271d755f6892d443786094d2607b505ff180ca817cb9eeabe1e5eR166-R187). @Kludex If you're interested, I can submit a PR updating the test here as well.

Reply:
Yeah... It was my bad. This test doesn't exist on master anymore.

Reply:
Oh indeed, [I found it](https://github.com/encode/starlette/pull/1962)